crates/transaction-pool/src/pool/best.rs

```diff
@@ -95,6 +95,13 @@ impl<T: TransactionOrdering> BestTransactions<T> {
     /// Mark the transaction and it's descendants as invalid.
     pub(crate) fn mark_invalid(&mut self, tx: &Arc<ValidPoolTransaction<T::Transaction>>) {
         self.invalid.insert(*tx.hash());
+
+        // Mark all descendants as invalid
+        let mut descendant_id = tx.id().descendant();
+        while let Some(descendant) = self.all.get(&descendant_id) {
+            self.invalid.insert(*descendant.transaction.hash());
+            descendant_id = descendant.transaction.id().descendant();
+        }
```

Comment:
we kinda already have an euqivalent check for this in `next`, so what we actually need to check instead is marking the `SenderId` as `self.invalid` and not the hash. so instead of `self.invalid.insert(*tx.hash());` we need to track the tx.sender_id instead